### ISSUE TITLE
Redeploying fargate atlantis task always breaks

### ISSUE THREAD

**Issue by d33psky** [terraform-aws-modules/terraform-aws-atlantis#383]
## Description

When fargate has an atlantis task running and a new deployment was issued the existing task prevents the new one from starting with this error : 

`Error: initializing server: starting BoltDB: timeout (a possible cause is another Atlantis instance already running
`

Note: EFS is being used.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Reproduction Code [Required]

Update terraform to redeploy the atlantis fargate task.

## Expected behavior

The new tasks starts and fargate stops the old task.

## Actual behavior

The new task fails with above BoltDB error which makes the new task fail, fargate retries starting a new task which also fails, etc.

### Terminal Output Screenshot(s)

`Error: initializing server: starting BoltDB: timeout (a possible cause is another Atlantis instance already running
`

## Workaround

After deploying an update to this module manually stop all running tasks and wait for fargate to start the new one.

**Comment by bryantbiggs** [comment]
we can't control this behavior in the module - its up to users

**Comment by mayn** [comment]
@bryantbiggs please consider reopening this as a bug or at minimum an opportunity to update docs for clarity. This occurs with the efs example from this module. It’s not really apparent until `wait_for_steady_state = true` at the service level or navigate to ecs task logs in AWS. 

Although atlantis itself and the use of boltdb emits the error, this module behavior is internally defaulting `deployment_minimum_healthy_percent = 66`.
1.  https://github.com/terraform-aws-modules/terraform-aws-atlantis/blob/3433ecf918056e3d347e774c81d3b878e51fbe18/main.tf#L236 

ECS will first launch new a task and the cutover. This works fine in when `enable_efs = false`, as the there is no shared filesystem. 

When `enable_efs = true` and an atlantis task beyond the initial deployment is provisioned , the deployment fails as described by the original author of this issue due to the shared filesystem + atlantis + boltdb

@d33psky , A work around, that seems to be working is to set `deployment_minimum_healthy_percent = 0` when `enable_efs = true`. This looks like it allows ECS to terminate the existing task, before launching the new deployment.
The cost is atlantis service is unavailable until the new deployment comes online. 

```
module “atlantis” {
    …
    # ECS service
    service = {
        deployment_minimum_healthy_percent = 0
    }
}

**Comment by bryantbiggs** [comment]
ayyy, some proper information!

**Comment by commarla** [comment]
I just hit the same issue and I confirm than setting this works fine with EFS (with a small downtime):
```
  service = {
    # we want only 1 instance running at the same time
    deployment_minimum_healthy_percent = 0
    deployment_maximum_percent         = 100
    
    ...
 }
  ```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.2.1 :tada:
